In [26]:
import pandas as pd

df = pd.read_json("../data/financebench_merged.jsonl", lines=True)

print(df.shape)
df.head()

(150, 15)


,financebench_id,company,doc_name,question_type,question_reasoning,domain_question_num,question,answer,justification,dataset_subset_label,evidence,gics_sector,doc_type,doc_period,doc_link
0,financebench_id_03029,3M,3M_2018_10K,metrics-generated,Information extraction,NaN,What is the FY2018 capital expenditure amount ...,$1577.00,The metric capital expenditures was directly e...,OPEN_SOURCE,[{'evidence_text': 'Table of Contents 3M Comp...,Industrials,10k,2018,https://investors.3m.com/financials/sec-filing...
1,financebench_id_04672,3M,3M_2018_10K,metrics-generated,Information extraction,NaN,Assume that you are a public equities analyst....,$8.70,"The metric ppne, net was directly extracted fr...",OPEN_SOURCE,[{'evidence_text': 'Table of Contents 3M Comp...,Industrials,10k,2018,https://investors.3m.com/financials/sec-filing...
2,financebench_id_00499,3M,3M_2022_10K,domain-relevant,Logical reasoning (based on numerical reasoning),dg06,Is 3M a capital-intensive business based on FY...,"No, the company is managing its CAPEX and Fixe...",CAPEX/Revenue\nFixed Assets/Total Assets\nROA=...,OPEN_SOURCE,[{'evidence_text': '3M Company and Subsidiarie...,Industrials,10k,2022,https://investors.3m.com/financials/sec-filing...
3,financebench_id_01226,3M,3M_2022_10K,domain-relevant,Logical reasoning (based on numerical reasonin...,dg17,What drove operating margin change as of FY202...,Operating Margin for 3M in FY2022 has decrease...,NaN,OPEN_SOURCE,"[{'evidence_text': 'SG&A, measured as a percen...",Industrials,10k,2022,https://investors.3m.com/financials/sec-filing...
4,financebench_id_01865,3M,3M_2022_10K,novel-generated,NaN,NaN,"If we exclude the impact of M&A, which segment...",The consumer segment shrunk by 0.9% organically.,NaN,OPEN_SOURCE,[{'evidence_text': 'Worldwide Sales Change By...,Industrials,10k,2022,https://investors.3m.com/financials/sec-filing...


In [27]:
df.columns.tolist()

['financebench_id',
 'company',
 'doc_name',
 'question_type',
 'question_reasoning',
 'domain_question_num',
 'question',
 'answer',
 'justification',
 'dataset_subset_label',
 'evidence',
 'gics_sector',
 'doc_type',
 'doc_period',
 'doc_link']

In [28]:
for i in range(5):
    print("=" * 80)
    print("Company:", df.loc[i, "company"])
    print("Document:", df.loc[i, "doc_name"])
    print("Question type:", df.loc[i, "question_type"])
    print("Reasoning:", df.loc[i, "question_reasoning"])
    print("Question:", df.loc[i, "question"])
    print("Answer:", df.loc[i, "answer"])

Company: 3M
Document: 3M_2018_10K
Question type: metrics-generated
Reasoning: Information extraction
Question: What is the FY2018 capital expenditure amount (in USD millions) for 3M? Give a response to the question by relying on the details shown in the cash flow statement.
Answer: $1577.00
Company: 3M
Document: 3M_2018_10K
Question type: metrics-generated
Reasoning: Information extraction
Question: Assume that you are a public equities analyst. Answer the following question by primarily using information that is shown in the balance sheet: what is the year end FY2018 net PPNE for 3M? Answer in USD billions.
Answer: $8.70
Company: 3M
Document: 3M_2022_10K
Question type: domain-relevant
Reasoning: Logical reasoning (based on numerical reasoning)
Question: Is 3M a capital-intensive business based on FY2022 data?
Answer: No, the company is managing its CAPEX and Fixed Assets pretty efficiently, which is evident from below key metrics:
CAPEX/Revenue Ratio: 5.1%
Fixed assets/Total Assets: 2

In [29]:
df["question_type"].value_counts()

question_type
metrics-generated    50
domain-relevant      50
novel-generated      50
Name: count, dtype: int64

In [30]:
df["question_reasoning"].value_counts()

question_reasoning
Numerical reasoning                                                                             43
Information extraction                                                                          31
Numerical reasoning OR Logical reasoning                                                         6
Logical reasoning (based on numerical reasoning)                                                 5
Logical reasoning (based on numerical reasoning) OR Logical reasoning                            5
Logical reasoning (based on numerical reasoning) OR Numerical reasoning OR Logical reasoning     4
Numerical reasoning OR information extraction                                                    4
Information extraction OR Logical reasoning OR                                                   1
Information extraction OR Logical reasoning                                                      1
Name: count, dtype: int64

In [31]:
print("Companies:", df["company"].nunique())
print("Documents:", df["doc_name"].nunique())

display(df["company"].value_counts())
display(df["gics_sector"].value_counts())
display(df["doc_type"].value_counts())
display(df["doc_period"].value_counts().sort_index())

Companies: 32
Documents: 84


company
PepsiCo                 11
Amcor                    9
Johnson & Johnson        9
3M                       8
AMD                      8
Best Buy                 8
Boeing                   8
American Express         7
MGM Resorts              7
Pfizer                   6
Ulta Beauty              6
Adobe                    5
JPMorgan                 5
Verizon                  5
Corning                  4
CVS Health               4
General Mills            4
Nike                     4
AES Corporation          3
Amazon                   3
American Water Works     3
Block                    3
Coca-Cola                3
Lockheed Martin          3
Walmart                  3
Activision Blizzard      2
Foot Locker              2
Microsoft                2
Netflix                  2
Costco                   1
Kraft Heinz              1
Paypal                   1
Name: count, dtype: int64

gics_sector
Consumer Discretionary    30
Consumer Staples          23
Information Technology    22
Industrials               19
Health Care               19
Financials                13
Communication Services     9
Materials                  9
Utilities                  6
Name: count, dtype: int64

doc_type
10k         112
10q          15
Earnings     14
8k            9
Name: count, dtype: int64

doc_period
2015     3
2016     3
2017     6
2018     7
2019     8
2020    10
2021    14
2022    61
2023    35
2024     3
Name: count, dtype: int64

In [32]:
df.loc[0, "evidence"]

[{'evidence_text': 'Table of Contents \n3M Company and Subsidiaries\nConsolidated Statement of Cash Flow s\nYears ended December 31\n \n(Millions)\n \n2018\n \n2017\n \n2016\n \nCash Flows from Operating Activities\n \n \n \n \n \n \n \nNet income including noncontrolling interest\n \n$\n5,363 \n$\n4,869 \n$\n5,058 \nAdjustments to reconcile net income including noncontrolling interest to net cash\nprovided by operating activities\n \n \n \n \n \n \n \nDepreciation and amortization\n \n \n1,488 \n \n1,544 \n \n1,474 \nCompany pension and postretirement contributions\n \n \n(370) \n \n(967) \n \n(383) \nCompany pension and postretirement expense\n \n \n410 \n \n334 \n \n250 \nStock-based compensation expense\n \n \n302 \n \n324 \n \n298 \nGain on sale of businesses\n \n \n(545) \n \n(586) \n \n(111) \nDeferred income taxes\n \n \n(57) \n \n107 \n \n 7 \nChanges in assets and liabilities\n \n \n \n \n \n \n \nAccounts receivable\n \n \n(305) \n \n(245) \n \n(313) \nInventories\n \n \n(50

In [33]:
df["num_evidence_passages"] = df["evidence"].apply(len)

df["num_evidence_passages"].value_counts().sort_index()

num_evidence_passages
1    115
2     31
3      4
Name: count, dtype: int64

In [34]:
df.isnull().sum().sort_values(ascending=False)

domain_question_num      100
question_reasoning        50
justification             50
financebench_id            0
company                    0
doc_name                   0
question_type              0
question                   0
answer                     0
dataset_subset_label       0
evidence                   0
gics_sector                0
doc_type                   0
doc_period                 0
doc_link                   0
num_evidence_passages      0
dtype: int64

In [35]:
df["question"].duplicated().sum()

0

In [36]:
df.groupby("doc_name").size().sort_values(ascending=False)

doc_name
BOEING_2022_10K               7
AMERICANEXPRESS_2022_10K      7
AMD_2022_10K                  7
PEPSICO_2022_10K              5
ULTABEAUTY_2023Q4_EARNINGS    4
                             ..
JPMORGAN_2023Q2_10Q           1
KRAFTHEINZ_2019_10K           1
LOCKHEEDMARTIN_2020_10K       1
LOCKHEEDMARTIN_2021_10K       1
WALMART_2020_10K              1
Length: 84, dtype: int64

## Key Findings

- The FinanceBench sample contains 150 question-answer examples across 32 companies and 84 financial documents.
- The dataset is evenly split across three question types: metrics-generated, domain-relevant, and novel-generated.
- Questions require a mix of information extraction, numerical reasoning, and logical reasoning.
- Most source documents are 10-K filings, with additional 10-Q, earnings, and 8-K documents.
- Each question includes grounded evidence from the underlying financial document, including page-level metadata.
- Some questions require multiple evidence passages, which is important for retrieval evaluation.
- There are no duplicate questions.
- Some metadata fields are intentionally missing depending on the question type.

## Test Split Consideration

Multiple questions can come from the same source document. Therefore, a purely random question-level train/test split could place questions from the same filing in both sets.

For a more realistic evaluation, we split the data by `doc_name`, so that all questions from the same financial document remain in the same partition. This would reduce leakage and better test whether the RAG system can retrieve and answer questions from unseen documents.

In [37]:
from sklearn.model_selection import train_test_split

docs = df["doc_name"].unique().tolist()

train_docs, test_docs = train_test_split(
    docs,
    test_size=0.2,
    random_state=42
)

train_df = df[df["doc_name"].isin(train_docs)]
test_df = df[df["doc_name"].isin(test_docs)]

print("Train questions:", len(train_df))
print("Test questions:", len(test_df))

shared_docs = set(train_df["doc_name"]) & set(test_df["doc_name"])
print("Shared documents:", shared_docs)

Train questions: 124
Test questions: 26
Shared documents: set()


In [38]:
print("Train documents:", len(train_docs))
print("Test documents:", len(test_docs))

Train documents: 67
Test documents: 17
